### PR TITLE
(PC-16258) feat(search): set category filter on search button press

### PR DESCRIPTION
**Pull request by voisinhugo**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-16258

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

File: src/features/search/pages/__tests__/Categories.test.tsx [hunk 1]
```diff
@@ -36,6 +37,35 @@ describe('Categories component', () => {
     const cinemaSelected = renderCategories().toJSON()
     expect(cinemaSelected).toMatchDiffSnapshot(allSelected)
   })
+
+  it('should navigate to result page when pressing search button', () => {
```
Comment (cedricls1) [inline src/features/search/pages/__tests__/Categories.test.tsx:41]: Je pense que tu pourrais avoir 2 tests :
1. Tu cliques sur toutes les catégories et tu fais rechercher (tu peux utiliser ton initialState et offerCategories [])
2. Tu cliques sur une catégorie et tu fais rechercher (tu peux utiliser ton initialState et offerCategories[la catégorie sur laquelle tu as cliqué])

Comment (voisinhugo) [reply src/features/search/pages/__tests__/Categories.test.tsx:41]: Oui en effet, c'est un edge case "toutes les catégories" 👍

**Review by cedricls1 — APPROVED** [review]
